### DISCUSSION TITLE
Graphql get products

### DISCUSSION THREAD

**Discussion by dyk9456** [swellstores/community#145] [Help] [answered]
Using graphql api How can I get all products that's within a subcategorie

**Comment by mustafahoda** [comment, marked as answer]
Hi @dyk9456 ,

You can use the following GraphQL Query to get products by providing the Category ID. This will work for both Category and Subcategories.

```
query getCategories {
	categoryById(id: "<CATEGORY_ID>"){
        topId
        name
        products {
            id
            name
        }
    }
}
```

**Reply by dyk9456** [reply]
thanks for that reply using this answer how can I add a limit and page 
since the above code only gets me the top 15 and when using this code `query getProductById {
 products(page:1, limit: 100){
    results{
      name
    }
	}
}`
it dosnt let me filter by category slug

**Reply by mustafahoda** [reply]
I see a few different questions in your response. I'll attempt to address them below.

- Can we provide a limit in the query

  - Does not seem possible and seems to be a gap in the GraphQL SDK. I'll get a confirmation

- How do you get category by slug?

 ``` 
  query slugTest {
  categoryBySlug(slug: "men") {
    parentId
    topId
    products {
        id
        name
    }
  }
}
```

**Reply by agsmartwolf** [reply]
I try to get `categoryBySlug` too but fetched products are limited to 15 and I can't fix it. I can't add limit and page parameters to `products`

**Reply by logeshswell** [reply]
@dyk9456 There is a gap in graphql SDK with the following:
* Pagination & accepting limit value is currently not supported with the nested collection.
* Filtering products by category slug involves nested querying which is also in our pipeline.

I will keep you posted if it's released.
